third-party/realdds/src/dds-stream-server.cpp

```diff
@@ -3,74 +3,58 @@
 
 #include <realdds/dds-stream-server.h>
 
+#include <realdds/dds-topic-writer.h>
+#include <realdds/dds-topic.h>
 #include <realdds/dds-participant.h>
 #include <realdds/dds-utilities.h>
-#include <realdds/topics/dds-topics.h>
+#include <realdds/topics/image/image-msg.h>
+#include <realdds/topics/image/imagePubSubTypes.h>
 
-#include <fastdds/dds/domain/DomainParticipant.hpp>
-#include <fastdds/dds/publisher/Publisher.hpp>
+#include <fastdds/dds/topic/Topic.hpp>
 #include <fastdds/dds/publisher/DataWriter.hpp>
-#include <fastdds/dds/publisher/DataWriterListener.hpp>
-#include <fastdds/dds/publisher/qos/PublisherQos.hpp>
-#include <librealsense2/utilities/concurrency/concurrency.h>
 
-#include <iostream>
 
 using namespace eprosima::fastdds::dds;
 using namespace realdds;
 
-dds_stream_server::dds_stream_server( std::shared_ptr< dds_participant > const & participant,
-                                      eprosima::fastdds::dds::Publisher * publisher,
-                                      const std::string & topic_root,
-                                      const std::string & stream_name )
-    : _participant( participant )
-    , _publisher( publisher )
-    , _topic( nullptr )
-    , _data_writer( nullptr )
+
+dds_stream_server::dds_stream_server( std::shared_ptr< dds_topic_writer > const & writer )
+    : _writer( writer )
 {
-    std::string topic_name = topics::device::image::construct_topic_name( topic_root, stream_name );
+}
 
-    eprosima::fastdds::dds::TypeSupport topic_type( new topics::device::image::type );
 
-    DDS_API_CALL( _participant->get()->register_type( topic_type ) );
-    _topic = DDS_API_CALL( _participant->get()->create_topic( topic_name, topic_type->getName(), TOPIC_QOS_DEFAULT ) );
+dds_stream_server::~dds_stream_server()
+{
+}
 
-    // TODO:: Maybe we want to open a writer only when the stream is requested?
-    DataWriterQos wqos = DATAWRITER_QOS_DEFAULT;
-    wqos.data_sharing().off();
-    wqos.reliability().kind = BEST_EFFORT_RELIABILITY_QOS;
-    wqos.durability().kind = VOLATILE_DURABILITY_QOS;
-    wqos.publish_mode().kind = SYNCHRONOUS_PUBLISH_MODE;
-    // Our message has dynamic size (unbounded) so we need a memory policy that supports it
-    wqos.endpoint().history_memory_policy = eprosima::fastrtps::rtps::PREALLOCATED_WITH_REALLOC_MEMORY_MODE;
 
-    _data_writer = DDS_API_CALL( _publisher->create_datawriter( _topic, wqos ) );
+std::shared_ptr< dds_topic > const& dds_stream_server::get_topic() const
+{
+    return _writer->topic();
 }
 
 
-dds_stream_server::~dds_stream_server()
+void dds_stream_server::start_streaming( const image_header & header )
```

Comment:
`start_streaming` is just setting a profile? name is confusing, thought it will start sending available frames from a queue.
Also looked for a matching `stop_streaming` :-)

Reply:
Right now the way to start streaming is via the device-server, and the actual stream-servers are not exposed at all.
This will likely change.
I also added a stop_streaming, but not in this PR.

Reply:
Also -- there will be no queue. No frames are "waiting" to be sent, unless we think that's needed. Is it? We can talk about it...

Reply:
(and certainly no frames will be waiting before we've actually started streaming)

Reply:
Queue is needed if we want a sender thread. Usually not needed